### DISCUSSION TITLE
PSA: Alder Lake S CPUs appear to have a fall through on Windows

### DISCUSSION THREAD

**Discussion by Pokeman2003** [vhelin/wla-dx#648] [General]
I don't think isn't exactly an issue you can fix, per se, which is why I haven't brought it up as an issue. 
In short, there is a fall through problem during a kernel call with WLA-DX on Intel's Alder Lake S architecture. I have two different machines that have them, and have confirmed that this happens on both. These machines have Alder Lake S 840 CPUs. 
The first attempt of executing after machine start will act somewhat normally but will end operations abruptly, and subsequent attempts will either fail to do anything or spit out complete gibberish + destabilize any running instance of command prompt.
Again, it is highly unlikely you can do anything to fix this. It happens during a kernel call, and it's skipping through a JNE instruction that it should be executing.

**Comment by vhelin** [comment]
For starters I don't have a computer with an Alder Lake S CPU and none of my friends or close relatives have one so I don't know how I could debug this...

Also this sounds like it affects a lot of other programs as well, maybe? How did you come to the conclusion that it's a bug in Alder Lake S? This is the first time I hear about a bug like that.

Do you know what in WLA DX uses that kernel call, and what call is it?

**Reply by Pokeman2003** [reply]
These bugs are pretty rare and near impossible to pin down, because they only occur during extraordinary edge cases that weren't accounted for. By all accounts, the machine code logic doesn't have a problem, it's literally a hardware glitch, which is why I said it's probably not even an issue you can fix.
For discovery, we tested on 7 machines. 3 of them older(pre-2012,) 4 of them newer(post-2018.) The two machines that had the issue were both Alder Lake S 840 machines. Unfortunately, I didn't think to mention this when we discovered the issue and tested it in October, otherwise I'd have way more information to give you.

I don't remember which kernel call it was, and only have memory dumps from when he dumped on my machine, but I think it had something to do with loading a file. I don't know why it wouldn't be affecting other software if that's the case though. If I recall correctly, it also occurred between both the linker and assembler, and it definitely happened early on so any kernel calls shared between the two would narrow it down.

I'm going to issue a correction: It does not affect the 840 chipset. That's a chipset from 2003-2004, and anyone who caught that before hand deserves to backhand me for making that assumption. It affects the Alder Lake S _840_. I have no idea what that number stands for, the information online is practically non-existent. I could only find it appearing in some PC listings, and it also appears in the hardware information dumps that we took of the computers having problems.

**Reply by vhelin** [reply]
Hmm, WLA DX uses pretty standard file operations that ANSI C provides. Could this issue be solved if you used a different C compiler to compile WLA DX? Or a different version of what you are currently using?

**Reply by Pokeman2003** [reply]
I'll give it a shot, but it'll take me a couple of weeks to get back to you, since I don't really have frequent contact with this guy.

**Comment by vhelin** [comment]
Maybe try to use latest Visual Studio as I guesstimate the compiler it has
should know about issues with Alder Lake S CPUs... ?
